pyiron_base/jobs/job/generic.py

```diff
@@ -1124,10 +1122,12 @@ def from_hdf(self, hdf=None, group_name=None):
         job_dict = self._hdf5.read_dict_from_hdf()
         with self._hdf5.open("input") as hdf5_input:
             job_dict["input"] = hdf5_input.read_dict_from_hdf(recursive=True)
-        self.from_dict(job_dict=job_dict)
-
+        # Backwards compatibility to the previous HasHDF based interface
         if "executable" in self._hdf5.list_groups():
-            self.executable.from_hdf(self._hdf5)
+            exe_dict = self._hdf5["executable/executable"].to_object().to_builtin()
+            exe_dict["READ_ONLY"] = self._hdf5["executable/executable/READ_ONLY"]
+            job_dict["executable"] = {"executable": exe_dict}
```

Comment:
This is triggered in both cases when on "old" executable or a "new" one is present, right?  Does `to_object().to_builtin()` really work in both cases?  Imo the preferable solution would be to keep the `Executable.from_hdf` around and have there logic that checks `__hdf_version__` and either does the old loading logic or `from_dict`s.

Reply:
No the important part is the `list_groups()` call. The previous implementation created an HDF group, the new one just a single node with the content of the node being a JSON dictionary.